bpython/curtsiesfrontend/repl.py

```diff
@@ -844,6 +846,13 @@ def on_control_d(self):
             self.current_line = (self.current_line[:self.cursor_offset] +
                                  self.current_line[(self.cursor_offset + 1):])
 
+    def on_control_o(self):
+        next_idx = -self.rl_history.index + 1
+        next = self.rl_history.entries[next_idx]
+        self.on_enter()
+        print("Next is : " + next)
+        self._set_current_line(next)
```

Comment:
The current code works well the first time, but not the second time; if trying to use ctrl-o to quickly re-enter a function longer than 2 lines for example. We should reset `self.rl_history.index` here too so that we're still back in history. (only a problem because `on_enter()` sets `self.rl_history.index` to 0).

Maybe something like
~~~
index = self.rl_history.index
...everything else in the method
self.rl_history.index = index
~~~